File: DailyLeetcode/5Aug/FruitIntoBaskets2.py
class Solution:
    def numOfUnplacedFruits(self, fruits: List[int], baskets: List[int]) -> int:
        unplaced=0
        n=len(baskets)
        is_used=[False]*n

        for fruit in fruits:
            is_placed = False
            for j in range(0,len(baskets),1):
                if fruit <= baskets[j] and not is_used[j]:
                    is_placed=True
                    is_used[j]=True
                    break

            if(is_placed==False):
                unplaced+=1

        return unplaced

File: DailyLeetcode/5Aug/FruitIntoBaskets2.cpp
//Time complexity=O(n^2)
//Space complexity=O(1)
class Solution {
public:
    int numOfUnplacedFruits(vector<int>& fruits, vector<int>& baskets) {
        int n=fruits.size();
        int cnt=0;
        for(auto fruit:fruits){
            int unset=1;
            for(int i=0;i<n;i++){
                if(fruit<=baskets[i]){
                    baskets[i]=0; //make that zero
                    unset=0; // and also don't count it again
                    break;
                }
            }
            cnt+=unset;//add to the cnt if it is not placed
        }
        return cnt;
        
    }
};